x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix_view.tsx

```diff
@@ -0,0 +1,318 @@
+  const [dynamicSize, setDynamicSize] = useState<boolean>(false);
+
+  // used to give the use the option to customize the fields used for the matrix axes
+  const [fields, setFields] = useState<string[]>([]);
+
+  useEffect(() => {
+    const defaultFields =
+      allFields.length > SCATTERPLOT_MATRIX_DEFAULT_FIELDS
+        ? allFields.slice(0, SCATTERPLOT_MATRIX_DEFAULT_FIELDS)
+        : allFields;
+    setFields(defaultFields);
+  }, [allFields]);
+
+  // the amount of documents to be fetched
+  const [fetchSize, setFetchSize] = useState<number>(SCATTERPLOT_MATRIX_DEFAULT_FETCH_SIZE);
+  // flag to add a random score to the ES query to fetch documents
+  const [randomizeQuery, setRandomizeQuery] = useState<boolean>(false);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  // contains the fetched documents and columns to be passed on to the Vega spec.
+  const [splom, setSplom] = useState<{ items: any[]; columns: string[] } | undefined>();
+
+  // formats the array of field names for EuiComboBox
+  const fieldOptions = useMemo(
+    () =>
+      allFields.map((d) => ({
+        label: d,
+      })),
+    [allFields]
+  );
+
+  const fieldsOnChange = (newFields: EuiComboBoxOptionOption[]) => {
+    setFields(newFields.map((d) => d.label));
+  };
+
+  const fetchSizeOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setFetchSize(
+      Math.min(
+        Math.max(parseInt(e.target.value, 10), SCATTERPLOT_MATRIX_DEFAULT_FETCH_MIN_SIZE),
+        SCATTERPLOT_MATRIX_DEFAULT_FETCH_MAX_SIZE
+      )
+    );
+  };
+
+  const randomizeQueryOnChange = () => {
+    setRandomizeQuery(!randomizeQuery);
+  };
+
+  const dynamicSizeOnChange = () => {
+    setDynamicSize(!dynamicSize);
+  };
+
+  const { euiTheme } = useCurrentEuiTheme();
+
+  useEffect(() => {
+    async function fetchSplom(options: { didCancel: boolean }) {
+      setIsLoading(true);
+      try {
+        const queryFields = [
+          ...fields,
+          ...(color !== undefined ? [color] : []),
+          ...(legendType !== undefined ? [] : [`${resultsField}.${OUTLIER_SCORE_FIELD}`]),
+        ];
+
+        const query = randomizeQuery
+          ? {
+              function_score: {
+                random_score: { seed: 10, field: '_seq_no' },
+              },
+            }
+          : { match_all: {} };
+
+        const resp: SearchResponse7 = await esSearch({
+          index,
+          body: {
+            fields: queryFields,
+            _source: false,
+            query,
+            from: 0,
+            size: fetchSize,
+          },
+        });
+
+        if (!options.didCancel) {
+          const items = resp.hits.hits.map((d) =>
+            getProcessedFields(d.fields, (key: string) =>
+              key.startsWith(`${resultsField}.feature_importance`)
+            )
+          );
+
+          setSplom({ columns: fields, items });
+          setIsLoading(false);
+        }
+      } catch (e) {
+        // TODO error handling
```

Comment:
Is this todo for a follow up?

Reply:
Yes, it just shows up as a new line because the file was moved and git didn't recognize it was a move. Better error handling is in the list of follow ups in the meta issue here #84420